registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/ABISHandlerUtil.java

```diff
@@ -119,31 +121,83 @@
+		List<AbisRequestDto> identifyRequests = getAllIdentifyRequest(registrationStatusDto.getRegistrationId());
+		// if there are unprocessed pending identify request for same transaction id then consider it as duplicate
+		if (isIdentifyRequestsPendingForLatestTransactionId(identifyRequests))
+			return AbisConstant.DUPLICATE_FOR_SAME_TRANSACTION_ID;
+		// else if all the identify requests processed for latest transaction id
+		else if (isAllIdentifyRequestProcessedForLatestTrnId(identifyRequests))
+			return AbisConstant.POST_ABIS_IDENTIFICATION;
+		else
 			return AbisConstant.PRE_ABIS_IDENTIFICATION;
-		}
-		return AbisConstant.POST_ABIS_IDENTIFICATION;
 	}
 
 	/**
-	 * Gets the matched reg ids.
+	 * This method returns all identify requests
 	 *
 	 * @param registrationId
 	 *            the registration id
 	 * @return the matched reg ids
 	 */
-	private List<AbisRequestDto> getMatchedRegIds(String registrationId) {
+	private List<AbisRequestDto> getAllIdentifyRequest(String registrationId) {
 		String latestTransactionId = utilities.getLatestTransactionId(registrationId);
 
 		List<String> regBioRefIds = packetInfoDao.getAbisRefMatchedRefIdByRid(registrationId);
 
-		List<AbisRequestDto> abisRequestDtoList = new ArrayList<>();
-
 		if (!regBioRefIds.isEmpty()) {
-			abisRequestDtoList = packetInfoManager.getInsertOrIdentifyRequest(regBioRefIds.get(0), latestTransactionId);
+			List<AbisRequestDto> abisRequestDtoList = packetInfoManager.getInsertOrIdentifyRequest(regBioRefIds.get(0), latestTransactionId);
+			if (!CollectionUtils.isEmpty(abisRequestDtoList)) {
+				return abisRequestDtoList.stream().filter(reqDto ->
+						reqDto.getRequestType().equalsIgnoreCase(AbisStatusCode.IDENTIFY.toString())).collect(Collectors.toList());
+			}
 		}
 
-		return abisRequestDtoList;
+		return null;
+	}
+
+	/**
+	 * This method returns all processed identify requests
+	 *
+	 * @param identifyRequests
+	 * @return
+	 */
+	private boolean isAllIdentifyRequestProcessedForLatestTrnId(List<AbisRequestDto> identifyRequests) {
+		boolean isProcesed = false;
+		// check if all identify requests are processed
+		if (!CollectionUtils.isEmpty(identifyRequests)) {
+			for (AbisRequestDto identifyReq : identifyRequests) {
+				if (!identifyReq.getStatusCode().equalsIgnoreCase(AbisStatusCode.PROCESSED.toString())) {
+					isProcesed = false;
+					break;
+				}
+				else
+					isProcesed = true;
+			}
+		}
+		return isProcesed;
+	}
+
+	/**
+	 * This method returns all unprocessed identify requests
+	 *
+	 * @param identifyRequests
+	 * @return
+	 */
+	private boolean isIdentifyRequestsPendingForLatestTransactionId(List<AbisRequestDto> identifyRequests) {
+		boolean isPending = false;
+		// check if all identify requests are processed
+		if (!CollectionUtils.isEmpty(identifyRequests)) {
+			for (AbisRequestDto identifyReq : identifyRequests) {
+				if (identifyReq.getStatusCode().equalsIgnoreCase(AbisStatusCode.PROCESSED.toString())) {
```

Comment:
do we need this method? U r checking the same as the previous method. can we not reuse the older method